src/test/java/com/rackspace/salus/monitor_management/services/MonitorManagementPolicyTest.java

```diff
@@ -966,6 +967,41 @@ public void testHandleMonitorPolicyEvent_removePolicy_butKeepMonitor() {
     assertTrue(monitorRepository.findByTenantIdAndPolicyId(tenantId, policyId).isEmpty());
   }
 
+  /**
+   * Receive a remove monitor policy event and process it for a tenant
+   * that was previously using the policy and its entry in monitor_policies having
+   * null as monitor id
+   *
+   */
+  @Test
+  public void testHandleMonitorPolicyEvent_removePolicy_null_monitor_id() {
+    String tenantId = RandomStringUtils.randomAlphanumeric(10);
+    UUID policyId = UUID.randomUUID();
+    UUID policyMonitorId = currentMonitor.getId();
+
+    // store a monitor for the tenant that is tied to the policy in the event
+    Monitor clonedMonitor = createMonitorForPolicyForTenant(tenantId, policyId);
+
+    List<MonitorPolicyDTO> list = List.of((MonitorPolicyDTO) new MonitorPolicyDTO()
+        .setMonitorId(null)
+        .setId(policyId));
+
+    doReturn(list).when(policyApi).getEffectiveMonitorPoliciesForTenant(anyString(), anyBoolean());
+
+    PageRequest pageRequest = PageRequest.of(0, 1000);
+
+    doReturn(Page.empty()).when(boundMonitorRepository).findAllByTenantIdAndMonitor_IdIn(tenantId, List.of(clonedMonitor.getId()), pageRequest);
+
+    MonitorPolicyEvent event = (MonitorPolicyEvent) new MonitorPolicyEvent()
+        .setMonitorId(policyMonitorId)
+        .setTenantId(tenantId)
+        .setPolicyId(policyId);
+    monitorManagement.handleMonitorPolicyEvent(event);
+
+    // policy monitor no longer exists on tenant
+    assertTrue(monitorRepository.findByTenantIdAndPolicyId(tenantId, policyId).isEmpty());
```

Comment:
Be sure to follow the pattern of assertions in the same test class. This class uses `assertThat` with hamcrest matchers, so this would be
```suggestion
    assertThat(monitorRepository.findByTenantIdAndPolicyId(tenantId, policyId), hasSize(0));
```